enrichments/logs/internal/mobile/groupingkey.go

```diff
@@ -0,0 +1,34 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package mobile
+
+import (
+	"crypto/sha256"
+	"encoding/hex"
+	"regexp"
+)
+
+func CreateGroupingKey(stacktrace string) string {
```

Comment:
Is this the same grouping algorithm being used in the legacy collector? I'm concerned the iOS version will always be unique, because the stacktrace string in iOS contains unique keys.

Reply:
I've just updated the logic to calculate a grouping key only for Java stacktraces, that way it won't affect Swift errors.

Reply:
We'll probably have to add a parser to pull out the crashing thread out of the iOS `stacktrace` to generate a grouping key.